### PR TITLE
Add build-id to assembly kernel builds

### PR DESCRIPTION
**Pull request by AlexBrownAMD**

### CODE REVIEW

**Review by yoichiyoshida — APPROVED** [review]

**Review by babakpst — APPROVED** [review]

**Comment by TorreZuk** [comment]
Assuming this may be wanted also as hotfix for 5.5 to match the source kernel build-id.

File: Tensile/KernelWriterAssembly.py [hunk 1]
```diff
@@ -250,6 +250,11 @@ def getCompileArgs(self, sourceFileName, objectFileName, *moreArgs, isa=None, wa
   def getLinkCodeObjectArgs(self, objectFileNames, coFileName, *moreArgs):
     rv = [globalParameters['AssemblerPath'],
           '-target', 'amdgcn-amd-amdhsa']
+    # Add build-id for builds with rocm 5.3+
+    compilerVer = globalParameters['HipClangVersion'].split(".")[:2]
+    compilerVer = [int(c) for c in compilerVer]
+    if len(compilerVer) >= 2 and (compilerVer[0] > 5 or (compilerVer[0] == 5 and compilerVer[1] > 2)):
+      rv += ["-Xlinker", "--build-id"]
```
Comment (nakajee) [inline Tensile/KernelWriterAssembly.py:257]: Do we need to specify this for os.name=="nt" case?
At line 120 of TensileCreateLibrary.py, parameters are set without using this function.
Should we add --build-id option separately?

**Comment by nakajee** [comment]
Why CI tests are failing?

File: Tensile/KernelWriterAssembly.py (hunk 1 above)
Comment (AlexBrownAMD) [reply Tensile/KernelWriterAssembly.py:257]: Updated the nt code to also call this function

**Pull request merged**